docs/user_guide/advanced_composition/nested_parallelization.md

```diff
@@ -0,0 +1,190 @@
+
+## Nested Dynamics
+
+This example shows how to break down a large workflow into smaller workflows and then compose them together to form a hierarchy. 
+
+```python
+"""
+The structure for 6 items and a chunk size of 2, the workflow will be broken down as follows:
+
+multi_wf -> level1 -> level2 -> core_wf -> step1 -> step2
+                             -> core_wf -> step1 -> step2
+                      level2 -> core_wf -> step1 -> step2
+                             -> core_wf -> step1 -> step2
+                      level2 -> core_wf -> step1 -> step2
+                             -> core_wf -> step1 -> step2
+"""
+
+import typing
+from flytekit import task, workflow, dynamic, LaunchPlan
+
+
+@task
+def step1(a: int) -> int:
+    return a + 1
+
+
+@task
+def step2(a: int) -> int:
+    return a + 2
+
+
+@workflow
+def core_wf(a: int) -> int:
+    return step2(a=step1(a=a))
+
+
+core_wf_lp = LaunchPlan.get_or_create(core_wf)
+
+
+@dynamic
+def level2(l: typing.List[int]) -> typing.List[int]:
+    return [core_wf_lp(a=a) for a in l]
+
+
+@task
+def reduce(l: typing.List[typing.List[int]]) -> typing.List[int]:
+    f = []
+    for i in l:
+        f.extend(i)
+    return f
+
+
+@dynamic
+def level1(l: typing.List[int], chunk: int) -> typing.List[int]:
+    v = []
+    for i in range(0, len(l), chunk):
+        v.append(level2(l=l[i:i + chunk]))
+    return reduce(l=v)
+
+
+@workflow
+def multi_wf(l: typing.List[int], chunk: int) -> typing.List[int]:
+    return level1(l=l, chunk=chunk)
+```
+
++++ {"lines_to_next_cell": 0}
+
+This shows a top-level workflow which uses 2 levels of dynamic workflows to process a list through some simple addition tasks and then flatten it again. Here is a visual representation of the execution in a Flyte console:
+
+:::{figure} https://raw.githubusercontent.com/flyteorg/static-resources/main/flytesnacks/user_guide/nested_parallel_top_level.png?raw=true
+:alt: Nested Parallelization UI View
+:class: with-shadow
+:::
+
+For each node in that top-level we can see 2 sub-workflows being run in parallel. 
+
+:::{figure} https://raw.githubusercontent.com/flyteorg/static-resources/main/flytesnacks/user_guide/nested_parallel_inner_dynamic.png?raw=true
+:alt: Inner Dynamic
+:class: with-shadow
+:::
+
+Finally, drilling into each sub-workflow, we'll see both those tasks being executed in series.
+
+:::{figure} https://raw.githubusercontent.com/flyteorg/static-resources/main/flytesnacks/user_guide/nested_parallel_subworkflow.png?raw=true
+:alt: Sub-Workflow
+:class: with-shadow
+:::
+
+## Mixed Parallelism
+
+This example is similar to the above but instead of using a dynamic to parallelize a 2-task serial workflow, we're using that workflow to call a map task which processes both inputs in parallel. This workflow has one fewer layers of parallelism so the outputs won't be the same, but it does still demonstrate how you can mix these different approaches to achieving concurrency.
```

Comment:
```suggestion
This example is similar to nested dynamic workflows, but instead of using a dynamic workflow to parallelize a core workflow with serial tasks, we use a core workflow to call a map task, which processes both inputs in parallel. This workflow has one less layer of parallelism, so the outputs won't be the same as those of the nested parallelization example, but it does still demonstrate how you can mix these different approaches to achieve concurrency.
```

(Let me know if this is an accurate rephrasing -- I assume the core workflow is what's calling the map task, but that seems like it might be slightly at odds with what you wrote.)

Reply:
Yeah "core" gets confused a bit once you change the number of layers but I still think it conveys what it needs to.